source/_docs/energy.markdown

```diff
@@ -21,3 +21,22 @@ Home Assistant is an open platform and so home energy management is not restrict
 If you have a sensor that returns instantaneous power readings (W or kW), then to add a sensor that returns energy usage or generation (kWh) refer to [Riemann sum integral integration](/integrations/integration/#energy)
 
 <img src='/images/docs/energy/energy-overview.png' alt='Visual representation of how all different energy forms relate.' style='border: 0;box-shadow: none;'>
+
+## Troubleshooting missing entities 
+
+### Condition
+
+You are trying to add a sensor to the energy dashboard, but it does not appear in the selection list. 
+
+### Remedy
+
+To find out why the sensor is not showing, check the following points:
+- The sensor must have the appropriate attributes. Check your entity attributes in **{% my developer_states title="Developer Tools > States" %}** to confirm the following:
+  - `device_class` must be `energy` for electricity grid, solar, or battery categories. It must be `gas` for gas, or `water` for water.
+  - `state_class` must be `total` or `total_increasing`.
+  - The sensor must have an appropriate `unit_of_measurement`. See the help text for each category to see which units are accepted. Units containing an exponent must match superscript characters exactly as defined, e.g. `m³` is accepted, `m3` is not.
+  
+  If any of the attributes are not correct, you can take it up with the integration developer that provides your sensor, or override missing attributes with [customize](/docs/configuration/customizing-devices/).
```

Comment:
This is should not! be done.

The right path to take it to report is with the integration.

Reply:
Should I remove reference to customize? I guess it can be dangerous if user does not understand what they are doing.

I only mention it because that's what I've had to do in some cases, like when building my own water meters. I used `riemann` to integrate water flow rate to total water consumption, but that integration doesn't have any concept of water device classes. So I had to customize to add device_class `water` to the riemann helper. 

I feel like this is potentially a helpful path, but if you think we should not mention it I can remove it.

Reply:
> Should I remove reference to customize?

Yes please 🙏 

>  I guess it can be dangerous if user does not understand what they are doing.

💯% 

>  I used riemann to integrate water flow rate to total water consumption, but that integration doesn't have any concept of water device classes

That sounds like an issue / think we should fix... Not work around...

Reply:
Reference to customize was removed.